tests/framework/data/BaseDataProviderTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace yiiunit\framework\data;
+
+use yiiunit\TestCase;
+use yii\data\BaseDataProvider;
+
+class BaseDataProviderTest extends TestCase
+{
+    public function testGenerateId()
+    {
+        $rc = new \ReflectionClass(BaseDataProvider::className());
+        $rp = $rc->getProperty('counter');
+        $rp->setAccessible(true);
+        $rp->setValue(null);
+
+        $this->assertNull((new ConcreteDataProvider())->id);
+        $this->assertNotNull((new ConcreteDataProvider())->id);
+    }
+
+}
+
+class ConcreteDataProvider extends BaseDataProvider
+{
+
+    /**
```

Comment:
I don't think PHPDoc is needed for overrided methods, especially in tests. Either use `@inheritdoc` or remove it completely. The latter is better IMO because we don't need to generate docs for tests suite.